server/sublist.go

```diff
@@ -236,8 +236,13 @@ func (s *Sublist) Match(subject string) *SublistResult {
 	s.Lock()
 	matchLevel(s.root, tokens, result)
 
+	// Store in cache and return to caller a copy of the results to avoid
+	// race when sub is removed from sublist and caller walks through the
+	// results.
+	cr := copyResult(result)
```

Comment:
This feels like it could be a performance hit for sure.

Reply:
I revised to just copy where needed (in matchLevel)